File: scrap-invoice-frontend/src/types/item.ts
// Tipe yang datang dari backend (misalnya saat fetch semua item)
export interface Item {
  id: number
  item_name: string
  item_category_id: number // ini biasanya ikut dikirim juga
  price_per_kg: number
  created_at: string
  updated_at: string
  category: {
    item_category_id: number
    item_category_name: string
  }
}

// Tipe untuk input form dari UI
export interface ItemFormInput {
  name: string
  category_id: number | null
  price_per_kg: number | null
}

// Tipe payload untuk API create/update (sesuai dengan struktur backend expect-nya)
export interface ItemFormPayload {
  item_name: string
  item_category: number
  price_per_kg: number
}

File: scrap-invoice-backend/models/item.go
package models

import "time"

type Item struct {
	ID             int       `gorm:"column:item_id;primaryKey" json:"id"`
	ItemName       string    `gorm:"column:item_name" json:"item_name"`
	ItemCategoryID int       `gorm:"column:item_category_id" json:"item_category"`
	PricePerKg     float64   `gorm:"column:price_per_kg" json:"price_per_kg"`
	CreatedAt      time.Time `gorm:"column:created_at" json:"created_at"`
	UpdatedAt      time.Time `gorm:"column:updated_at" json:"updated_at"`

	Category ItemCategory `gorm:"foreignKey:ItemCategoryID;references:ID" json:"category"`
}
